### PR TITLE
Integrated the storage layer of inbox Notes in the Yosemite layer

### LINKED ISSUE #6126: Integrate stored Inbox Notes in Yosemite layer
After having implemented the [Yosemite layer](https://github.com/woocommerce/woocommerce-ios/pull/6097) and the [Storage layer](https://github.com/woocommerce/woocommerce-ios/pull/6121) separately, now we should integrate the Storage layer with the existing Yosemite layer for Inbox Notes.

### PR DESCRIPTION
**Pull request by pmusolino**
Closes #6126 

### Description
After having implemented the [Yosemite layer](https://github.com/woocommerce/woocommerce-ios/pull/6097) and the [Storage layer](https://github.com/woocommerce/woocommerce-ios/pull/6121) separately, in this PR I integrated the work done on the Storage layer with the existing Yosemite layer for Inbox Notes. So, now every request fired in the Yosemite layer store the inbox notes locally in core data.

### Testing instructions
Just CI, the code is still not in use.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Yosemite/Yosemite/Actions/InboxNotesAction.swift [hunk 1]
```diff
@@ -8,11 +8,11 @@ public enum InboxNotesAction: Action {
...
-                           pageNumber: Int,
-                           pageSize: Int,
-                           orderBy: InboxNotesRemote.OrderBy,
-                           type: [InboxNotesRemote.NoteType]?,
-                           status: [InboxNotesRemote.Status]?,
+                           pageNumber: Int = Default.pageNumber,
+                           pageSize: Int = Default.pageSize,
+                           orderBy: InboxNotesRemote.OrderBy = .date,
+                           type: [InboxNotesRemote.NoteType]? = nil,
+                           status: [InboxNotesRemote.Status]? = nil,
```
Comment (jaclync) [inline Yosemite/Yosemite/Actions/InboxNotesAction.swift:15]: that's nice we can set a default value to an enum case's associated value now!

File: Yosemite/Yosemite/Stores/InboxNotesStore.swift [hunk 2]
```diff
@@ -71,7 +100,65 @@ private extension InboxNotesStore {
...
+    func upsertStoredInboxNotes(readOnlyInboxNotes: [Networking.InboxNote],
+                                in storage: StorageType,
+                                siteID: Int64) {
+        for inboxNote in readOnlyInboxNotes {
+            let storageInboxNote: Storage.InboxNote = {
+                if let storedInboxNote = storage.loadInboxNote(siteID: siteID, id: inboxNote.id) {
+                    return storedInboxNote
+                }
+                return storage.insertNewObject(ofType: Storage.InboxNote.self)
+            }()
```
Comment (jaclync) [inline Yosemite/Yosemite/Stores/InboxNotesStore.swift]: nit: could this be simplified like
```suggestion
            let storageInboxNote = storage.loadInboxNote(siteID: siteID, id: inboxNote.id) ?? storage.insertNewObject(ofType: Storage.InboxNote.self)
```

File: Yosemite/YosemiteTests/Stores/InboxNotesStoreTests.swift [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+            self.store.onAction(action)
+        }
+
+
+        // Then the initial inbox note should have it's values updated
+        let updatedInboxNote = viewStorage.loadInboxNote(siteID: sampleSiteID, id: 296)
+        XCTAssertTrue(result.isSuccess)
+        XCTAssertNotEqual(initialInboxNote.name, updatedInboxNote?.name)
+        XCTAssertNotEqual(initialInboxNote.title, updatedInboxNote?.title)
+        XCTAssertNotEqual(initialInboxNote.content, updatedInboxNote?.content)
```
Comment (jaclync) [inline Yosemite/YosemiteTests/Stores/InboxNotesStoreTests.swift:111]: nit: maybe we can assert on the `actions` to test the relationship?

File: Yosemite/Yosemite/Stores/InboxNotesStore.swift [hunk 4]
```diff
@@ -71,7 +100,65 @@ private extension InboxNotesStore {
...
+
+            storageInboxNote.update(with: inboxNote)
```
Comment (jaclync) [inline Yosemite/Yosemite/Stores/InboxNotesStore.swift:146]: we might also want to handle the `actions` relationship here? in `InboxNote+ReadOnlyConvertible`, only properties  are set

File: Yosemite/YosemiteTests/Stores/InboxNotesStoreTests.swift [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+            }
+
+            let action = InboxNotesAction.dismissInboxNote(siteID: self.sampleSiteID, noteID: sampleInboxNoteID) { result in
+                promise(result)
+            }
+            self.store.onAction(action)
+        }
+
+        // Then a valid inbox note should be stored
+        XCTAssertEqual(storedInboxNotesCount, 1)
```
Comment (jaclync) [inline Yosemite/YosemiteTests/Stores/InboxNotesStoreTests.swift:178]: nit: does any property change after dismissing a note? if so, maybe we can assert on that change as well?

File: Yosemite/YosemiteTests/Stores/InboxNotesStoreTests.swift [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+            }
+            self.store.onAction(action)
+        }
+
+        // Then the initial inbox action should have it's values updated
+        let updatedInboxNote = viewStorage.loadInboxNote(siteID: sampleSiteID, id: 296)
+        XCTAssertTrue(result.isSuccess)
+        XCTAssertNotEqual(initialInboxNote.actions.first?.name, updatedInboxNote?.actions?.first?.name)
+        XCTAssertNotEqual(initialInboxNote.actions.first?.label, updatedInboxNote?.actions?.first?.label)
+        XCTAssertNotEqual(initialInboxNote.actions.first?.status, updatedInboxNote?.actions?.first?.status)
```
Comment (jaclync) [inline Yosemite/YosemiteTests/Stores/InboxNotesStoreTests.swift:253]: nit: is it possible to assert with the expected value after a note is marked as actioned, like the `status` from `unactioned` to `actioned`?

File: Yosemite/YosemiteTests/Stores/InboxNotesStoreTests.swift [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+            }
+
+            let action = InboxNotesAction.markInboxNoteAsActioned(siteID: self.sampleSiteID, noteID: sampleInboxNoteID, actionID: sampleActionID) { result in
+                promise(result)
+            }
+            self.store.onAction(action)
+        }
+
+        // Then no inbox notes should be stored
+        XCTAssertEqual(storedInboxNotesCount, 0)
```
Comment (jaclync) [inline Yosemite/YosemiteTests/Stores/InboxNotesStoreTests.swift:277]: nit: maybe we can assert that the action's `status` is not changed afterward?

**Review by jaclync — APPROVED** [review]
Mostly a question about handling `actions`, then LGTM! Also some suggestions on assertions for your consideration 🙂

File: Yosemite/Yosemite/Stores/InboxNotesStore.swift (hunk 4 above)
Comment (pmusolino) [reply Yosemite/Yosemite/Stores/InboxNotesStore.swift:146]: 👀 👍 7d84bcea08043d9cad4e130798ef4e4d4583e9bc

**Comment by pmusolino** [comment]
Thank you @jaclync for the deep review 💯 I updated the code following your suggestions.
 I'm going to merge it after the CI is ✅

**Pull request merged**